src/java/test/org/apache/zookeeper/test/StaticHostProviderTest.java

```diff
@@ -117,8 +116,32 @@ public void testTwoInvalidHostAddresses() {
         list.add(new InetSocketAddress("a", 2181));
         list.add(new InetSocketAddress("b", 2181));
         new StaticHostProvider(list);
+	}
+
+    @Test
+    public void testReResolving() {
+        byte size = 1;
+        ArrayList<InetSocketAddress> list = new ArrayList<InetSocketAddress>(size);
+
+        // Test a hostname that resolves to multiple addresses
+        list.add(InetSocketAddress.createUnresolved("www.apache.org", 1234));
```

Comment:
I'm wondering if it's possible to mock this out? It would be great if our unit tests were not dependent on some other infrastructure.

Reply:
Good point I spent hours finding a way for it. Unfortunately static members can't be mocked with Mockito. We should use PowerMock for it (as described here https://blog.codecentric.de/en/2016/03/junit-testing-using-mockito-powermock/), but that'd be a new dependency not sure if it's acceptable here.

Reply:
Yeah, this is annoying. Although, another possibility would be to put the dns calls in a method and then subclass `StaticHostProvider` for the tests and overriding this method to return what you want.

Reply:
Makes sense, I'm working on it.

Reply:
This is done.